tracex/extraction/static/extraction/js/progress-bar.js

```diff
@@ -0,0 +1,53 @@
+const execute_button = document.getElementById('execute_button');
+const progress_box = document.getElementById('progress_box');
+
+function updateProgressBar() {
+    $.ajax({
+        url: '/extraction/filter/',
+        method: "GET",
+        success: function(data) {
+            // Update the progress bar element with the received progress data
+            const percentage = data.progress !== null ? data.progress + "%" : "0%";
+            const current_module = data.status;
+            progress_box.innerHTML = `
+            <div class="progress">
+                <div class="progress-bar progress-bar-striped progress-bar-animated ${getProgressBarColor(data.progress)}" role="progressbar" style="width: ${percentage};" aria-valuenow="${percentage}" aria-valuemin="0" aria-valuemax="100">${percentage}</div>
+            </div>
+            ${current_module ? `<div class="progress-container"><p>${current_module} is currently running</p></div>` : '<div class="progress-container"><p>Execute Extraction Pipeline</p></div>'}
+        `;
+
+            // If the task is not complete, continue checking for progress // hier muss data.progress rein
+            if (data.progress === 100) {
+                // Hide the progress bar and show the result button
+                execute_button.classList.add('not_visible');
+                
+            } else {
+                // If progress is not 100, continue fetching progress
+                setTimeout(updateProgressBar, 1000); // Fetch progress again after 1 second
+            }
+        },
+        error: function() {
+            console.log("Error occurred while fetching progress.");
+        }
+    });
+}
+
+function getProgressBarColor(progress) {
```

Comment:
Maybe just change the progress bar to one color. There color has no indication for us and colors like "bg-danger" do not sound very suitable in our case.